src/Rule/Path.php

```diff
@@ -86,7 +86,7 @@ public function __invoke(ServerRequestInterface $request, Route $route)
 
         foreach ($this->route->tokens as $name => $pattern) {
             if (is_callable($pattern)) {
-                if (!$pattern($attributes[$name], $route, $request)) {
+                if (!$pattern($attributes[$name] ?? NULL, $route, $request)) {
```

Comment:
For PSR-2, `NULL` should be `null`

https://www.php-fig.org/psr/psr-2/ - section 2.5

Reply:
Good catch.